### ISSUE TITLE
[BUG] NiftiSpheresMasker - Combination of percent signal change normalization and high-pass filtering only outputs signals in incorrect range.

### ISSUE THREAD

**Issue by alexsayal** [nilearn/nilearn#4087]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [X] Linux
- [X] Mac
- [ ] Windows

### Operating system version

- Linux Ubuntu 18.04
- Mac OS Ventura 13.6

### Python version

- [ ] 3.12
- [ ] 3.11
- [ ] 3.10
- [X] 3.9
- [ ] 3.8

### nilearn version

0.10.2

### Expected behavior

Percent signal change (PSC) normalization method in NiftiSpheresMasker in combination with high-pass filtering should return reasonable PSC values.

### Current behavior & error messages

When extracting time courses with NiftiSpheresMasker, the combination of percent signal change normalization with high-pass filtering only outputs signals in an incorrect range.

I am using this example code https://nilearn.github.io/stable/auto_examples/03_connectivity/plot_sphere_based_connectome.html to showcase this.

This is the output image of the code snippet below. Please note the y-axis values range.
![image](https://github.com/nilearn/nilearn/assets/9429033/a82a0302-b886-4007-9298-939ecdfe532f)

### Steps and code to reproduce bug

```python
from nilearn import datasets

dataset = datasets.fetch_development_fmri(n_subjects=1)

dmn_coords = [(0, -52, 18), (-46, -68, 32)]
labels = [
    "Posterior Cingulate Cortex",
    "Left Temporoparietal junction",
]

from nilearn.maskers import NiftiSpheresMasker

masker = NiftiSpheresMasker(
    dmn_coords,
    radius=8,
    #detrend=True,
    standardize="psc",
    #standardize_confounds="zscore_sample",
    #low_pass=0.1,
    high_pass=0.01,
    t_r=2,
    memory="nilearn_cache",
    memory_level=1,
    verbose=2,
    clean__butterworth__padtype="even",  # kwarg to modify Butterworth filter
)

# Additionally, we pass confound information to ensure our extracted
# signal is cleaned from confounds.

func_filename = dataset.func[0]
confounds_filename = dataset.confounds[0]

time_series = masker.fit_transform(
    func_filename, confounds=[confounds_filename]
)

import matplotlib.pyplot as plt

for time_serie, label in zip(time_series.T, labels):
    plt.plot(time_serie, label=label)

plt.title("Default Mode Network Time Series")
plt.xlabel("Scan number")
plt.ylabel("Normalized signal")
plt.legend()
plt.tight_layout()
```

**Comment by bthirion** [comment]
Thx for reporting !
I do not have time to check now, but I guess that the things goes back to preprocessing steps that are not run in the right order. Would you be interested in diagnosing that more precisely ?

**Comment by alexsayal** [comment]
Hi! I am afraid I won't be able to check this soon, so please take over whenever you can 👍 Thank you!

**Comment by bthirion** [comment]
Well, indeed a high-pass filter removes the low-frequency signal, including the constant part. Maybe there is inconsistency among the processing options ?

**Comment by alexsayal** [comment]
As the high-pass filter seems to remove the constant part, the true mean of the signal is lost and hence the standardization is performed using a mean value very close to zero, which in turn results in those inflated values. Does it make sense to invert the order of the steps? Or set the filter to keep the constant part of the signal?

**Comment by bthirion** [comment]
The idea would be to save the signal mean before filtering, I guess.

**Comment by ymzayek** [comment]
> The idea would be to save the signal mean before filtering, I guess.

With this change we get:

![image](https://github.com/nilearn/nilearn/assets/63292494/b7c7cd7c-347c-40d1-bc6e-5bf2aa0d8905)

**Comment by htwangtw** [comment]
PSC method might need a deeper review.
I did some digging - As long as there's a mean small enough, we will get wild range and butterworth filtering is just one method that happens to consistently generate a mean close to zero. This includes the current test simulated signal (we generate the mean signal from a distribution of mu=0 and sigma=1).
